projects/packages/analyzer/src/Invocations/class-visitor.php

```diff
@@ -56,20 +52,56 @@ public function enterNode( Node $node ) {
 			} elseif ( $node->name instanceof Node\Expr\MethodCall ) {
 				$this->enterNode( $node->name );
 				$function_name = $node->name->var->name . '->' . $node->name->name->name;
+			} elseif ( $node->name instanceof Node\Expr\Include_ ) {
+				// calls like this: ( require "$root_dir/modules.php" )( $root_dir ).
+				$function_name = ''; // do nothing :shrug:.
 			} else {
 				$function_name = implode( '\\', $node->name->parts );
 			}
 
-			$this->invocations->add( new Function_Call( $this->file_path, $node->getLine(), $function_name ) );
+			$out = new Function_Call( $this->file_path, $node->getLine(), $function_name );
 		} elseif ( $node instanceof Node\Expr\ClassConstFetch ) {
-			$this->invocations->add(
-				new Static_Const( $this->file_path, $node->getLine(), Utils::node_to_class_name( $node->class ), $node->name->name )
-			);
+			$out = new Static_Const( $this->file_path, $node->getLine(), Utils::node_to_class_name( $node->class ), $node->name->name );
 		} else {
 			// print_r( $node );
 		}
+
+		if ( null !== $out ) {
+			$this->invocations->add( $out );
+		}
 	}
 
+	/**
+	 * Leave node handler
+	 *
+	 * @param Node $node node object.
+	 */
 	public function leaveNode( Node $node ) {
 	}
+
+	/**
+	 * Recursively generates a (multidimentional) array invocation token.
+	 *
+	 * @param Node $node node object.
+	 * @return string
+	 */
+	private function getArrayToken( Node $node ) {
```

Comment:
What's the difference between this and `Utils::node_to_class_name()`? It seems like you're trying to do basically the same thing, just with handling of different edge cases (that better handles complex stuff in the 'dimension' part, while this handles complex stuff in the 'var' part).

For that matter, "node_to_class_name" doesn't seem to really describe what that method is doing. There are some bits that seem like they're handling a "class name", but other bits that seem to be generically stringifying a variable reference expr.